src/sagemaker/workflow/function_step.py

```diff
@@ -101,6 +102,12 @@ def __init__(
 
         self.__job_settings = None
 
+        # It's for internal usage to retrieve execution id from the properties.
+        # However, we won't expose the properties of function step to customers.
+        self._properties = Properties(
```

Comment:
Currently, does the backend logic handle this for selective executions ? If I provide a properties object to the validator, I assume there will be an error thrown since (IIRC) we have only used Parameters and Executions Variables in the JSON get function.

Reply:
No, the backend does not handle the case (i.e. case 3 I mentioned in the overview) as you said.
That's why I mentioned at the top that the case 3 support requires a backend change.

This PR would enable support for case 1 and 2 which do not relate to the logic you mentioned

Reply:
Sweet ok was just double checking! Nice.